components/Navbar.tsx

```diff
@@ -29,7 +29,10 @@ export default function Navbar() {
     document.body.style.overflow = "auto";
   };
 
-  const [newLinkModal, setNewLinkModal] = useState(false);
+  // If the URL has a query parameter "link", open the new link modal
+  const shareUrl = new URL(window.location.href).searchParams.get("link");
+
+  const [newLinkModal, setNewLinkModal] = useState(!!shareUrl);
   const [newCollectionModal, setNewCollectionModal] = useState(false);
   const [uploadFileModal, setUploadFileModal] = useState(false);
```

Comment:
This may not affect the functionality of the application but it may be a good idea to clean up the URL after consuming the parameters.

Something like this example:

```suggestion
  const [uploadFileModal, setUploadFileModal] = useState(false);
  
  useEffect(() => {
    window.history.replaceState({}, document.title, window.location.pathname)
  }, [newLinkModal])
```